synapse/metrics/_legacy_exposition.py

```diff
@@ -219,11 +220,16 @@ def do_GET(self) -> None:
         except Exception:
             self.send_error(500, "error generating metric output")
             raise
-        self.send_response(200)
-        self.send_header("Content-Type", CONTENT_TYPE_LATEST)
-        self.send_header("Content-Length", str(len(output)))
-        self.end_headers()
-        self.wfile.write(output)
+        try:
+            self.send_response(200)
+            self.send_header("Content-Type", CONTENT_TYPE_LATEST)
+            self.send_header("Content-Length", str(len(output)))
+            self.end_headers()
+            self.wfile.write(output)
+        except BrokenPipeError as e:
+            logger.warning(
+                "BrokenPipeError when serving metrics (%s). Did Prometheus restart?", e
+            )
```

Comment:
I think a warning is the right logging level to use here.

But do we know what's causing the socket to close before we send the response on matrix.org?

Reply:
I would guess the prometheus scraper is occasionally restarted? Pure speculation though.